fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneOnlineIndexingTest.java

```diff
@@ -145,6 +153,115 @@ void luceneOnlineIndexingTestSimple() {
         assertTrue(allFiles.length < 12);
     }
 
+    @Test
+    void luceneOnlineIndexingTestWithRecordUpdates() throws IOException {
+        final Map<String, String> options = Map.of(
+                INDEX_PARTITION_BY_FIELD_NAME, "timestamp",
+                PRIMARY_KEY_SEGMENT_INDEX_V2_ENABLED, "true",
+                INDEX_PARTITION_HIGH_WATERMARK, String.valueOf(2));
+
+        Index index = complexPartitionedIndex(options);
+        disableIndex(index, COMPLEX_DOC);
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 2)
+                .build();
+
+        long startTime = System.currentTimeMillis();
+        try (final FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            recordStore.saveRecord(createComplexDocument(1623L, WAYLON, 1, startTime));
+            recordStore.saveRecord(createComplexDocument(1547L, WAYLON, 1, startTime + 1000));
+            recordStore.saveRecord(createComplexDocument(2222L, WAYLON + " who?", 1, startTime + 2000));
+            recordStore.saveRecord(createComplexDocument(899L, ENGINEER_JOKE, 1, startTime + 3000));
+            context.commit();
+        }
+
+        // do not index any records and leave the index writeOnly
+        try (final FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            final RuntimeException stopBuildException = new RuntimeException("stop build");
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setRecordStore(recordStore)
+                    .setIndex(index)
+                    .setInitialLimit(1)
+                    .setConfigLoader(config -> {
+                        throw stopBuildException;
```

Comment:
Another test that would probably be good to add, and be a bit more functional, would be to basically have a config loader that opens a context, and updates all the records in the index.
If you have it update them with some sort of counter, you can then search at the end once the index is readable and validate that they all have the right value. 
I think you'll also need to call `setMaxLimit(1)` to ensure that you modify the records between each record being updated.